pkg/client/client.go

```diff
@@ -299,11 +288,31 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, allAddresses,
 				if i+1 != len(sr.KubeProxyReplacement.Devices) {
 					devices += ", "
 				}
-
 			}
 
-			fmt.Fprintf(w, "KubeProxyReplacement:\t%s\t[%s]\t[%s]\n",
-				sr.KubeProxyReplacement.Mode, devices, strings.Join(features, ", "))
+			fmt.Fprintf(w, "Protocols: \t%s\n", strings.Join(sr.KubeProxyReplacement.Protocols, ","))
+
+			fmt.Fprintf(w, "Devices: \t[%s]\n", devices)
+
+			fmt.Fprintf(w, "Mode: \t%s\n", sr.KubeProxyReplacement.NodePortMode)
+
+			fmt.Fprintf(w, "XDP Acceleration: \t%s\n", sr.KubeProxyReplacement.NodePortAcceleration)
+
+			fmt.Fprintf(w, "Session affinity: \t%s\n", getStateStr(sr.KubeProxyReplacement.SessionAffinity))
+
+			fe := sr.KubeProxyReplacement.Features
+
+			fmt.Fprint(w, "Services: \n")
+
+			fmt.Fprintf(w, "\tClusterIP: \t%s\n", getStateStr(fe.ClusterIP.Enabled))
+
+			fmt.Fprintf(w, "\tLoadBalancer: \t%s\n", getStateStr(fe.LoadBalancer.Enabled))
+
+			fmt.Fprintf(w, "\tHostPort: \t%s\n", getStateStr(fe.HostPort.Enabled))
+
+			fmt.Fprintf(w, "\tExternalIPs: \t%s\n", getStateStr(fe.ExternalIPs.Enabled))
+
+			fmt.Fprintf(w, "\tNodePort: \t%s\n", getNodePortStateStr(fe.NodePort))
```

Comment:
Hmm. With the amount of configuration state here (admittedly partially auto-configured by the daemon) I wonder if `cilium status` is getting away from the original intent of reporting the "status" / health of the daemon. So far, several of the sections of `cilium status` are written such that by default they're quiet with a one-line status, then if there's anything failing, that will always be printed potentially over multiple lines, then for detailed information, you pass `--verbose` (or one of the individual flags to get additional information there). In sysdump for instance we always pass `--verbose` to ensure we have the maximum information.

One concern I have is that this new set of lines is almost as big as the entire status output today. That's double the amount of lines to read through to get a high-level status of subsystems.

I see the value in highlighting which configuration options are in use when debugging a datapath problem, but when I want a simple overview of the status of the daemon, I don't care so much about these details. These things seem almost more appropriate to fit into `cilium config`.

@borkmann I guess you're arguing that it's a common problem that someone reports something isn't working and there's no way to just say "provide me your datapath configuration". If you could ask "run `cilium status` and give me the output" with this version, you would more easily be able to reason about what's going on underneath. Is that your intent? Do you think it's reasonable to provide another way to fetch all datapath configuration after we initially establish from `cilium status` that we would need that particular information?

Reply:
:+1: for `cilium config`.

Reply:
I would like this information to be displayed without having to pass the `--verbose` flag, but I agree that there might be a better place to show the details.

Reply:
The more I think about this, the more I think the overall `cilium status` should be consistently formatted like:

```
subsystemA: <minimal healthy info>
subsystemB: <minimal healthy info>
subsystemC: <maybe I'm unhealthy...>
  <unhealthy status header table line>
  <name> <status> <other-fields>
  <name> <status> <other-fields>
  <(up to 10 items with line prompting user to use --verbose if there are more)>
subsystemD: ...
```

From that perspective, printing the individual configuration sub-options of kube-proxy replacement by default doesn't make sense to me. It's extra information that we may/may not need, that makes answering the question "Is Cilium healthy?" harder because you need to sort through more information that's describing the state of the system but not how to interpret it.

For the one-liner summary, I like the "partial" / "strict" / "probe" piece of information as this describes that the subsystem/feature is enabled (I'd probably consider `kubeProxyReplacement` as a dedicated subsystem by this point just given its complexity). I like the mention of devices, because it's fairly new for us to add eBPF programs on native devices (rather than just the container devices) and I'm still not sure exactly which options will automatically configure this or not. Knowing that "traffic on eth0 is subject to Cilium eBPF programs and therefore I can run `cilium monitor` / `hubble` to introspect" is useful.

Beyond that, if we're starting to debug an issue that might be related to kube-proxy replacement, it doesn't seem like it makes a big difference to me to print the configuration by default in `cilium status`. In an ideal case, KPR just works so we don't need to debug it. If we do need to debug it, then either we should get a sysdump ASAP so we can debug with the full cluster information, or we *can* perform command-by-command investigation. But in the last case, we are almost guaranteed to need more information than what `cilium status` provides, so it doesn't seem like a big deal to me to just run either one extra command or pass a `--kube-proxy-replacement-status` / `--verbose` flag,  or run `cilium config` to gather that specific info.

Reply:
+1 for kubeProxyReplacement as a dedicated subsystem
I feel like if we were to move the details under `cilium config`, it would be subjected to the same argument as that of `cilium status`.

Reply:
By "same argument" do you mean you're also concerned about verbosity there?

In general I'm less concerned in `cilium config`, because it's specifically about configuration and not about agent health.